src/App.tsx

```diff
@@ -1,223 +1,102 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
-import { AuthContext } from './components/Auth/AuthContext';
+import React, {
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import { AuthContext, AuthProvider } from './components/Auth/AuthContext';
+import { getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import { ErrorType } from './types/ErrorType';
+import { Filter } from './types/Filter';
+
+import { Header } from './components/Header/Header';
+import { TodoList } from './components/TodosList/TodosList';
+import { Footer } from './components/Footer/Footer';
+import { Error } from './components/Error/Error';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
+  const [todos, setTodos] = useState<Todo[] | []>([]);
```

Comment:
You don't need to add an empty array to the types. SomeType[] always covers the empty array as well
```suggestion
  const [todos, setTodos] = useState<Todo[]>([]);
```